comp/core/flare/helpers/send_flare.go

```diff
@@ -254,13 +272,65 @@ func SendTo(cfg pkgconfigmodel.Reader, archivePath, caseID, email, apiKey, url s
 		return "", err
 	}
 
-	r, err := readAndPostFlareFile(archivePath, caseID, email, hostname, url, source, client, apiKey)
-	if err != nil {
-		return "", err
+	// Retry logic for the actual flare file posting
+	var lastErr error
+	var maxTries = 3
+	var baseDelay = 100 * time.Millisecond
+
+	for attempt := 1; attempt <= maxTries; attempt++ {
+		if attempt > 1 {
+			// Exponential backoff
+			delay := time.Duration(attempt-1) * baseDelay
+			time.Sleep(delay)
+		}
+
+		r, err := readAndPostFlareFile(archivePath, caseID, email, hostname, url, source, client, apiKey)
+		if err != nil {
+			// Always close the response body if it exists
+			if r != nil && r.Body != nil {
+				r.Body.Close()
+			}
+
+			statusCode := 0
+			if r != nil {
+				statusCode = r.StatusCode
+			}
+			lastErr = err
+			if attempt < maxTries && isRetryableFlareError(err, statusCode) {
+				continue
+			}
+			// If it's not retryable, return immediately
+			if !isRetryableFlareError(err, statusCode) {
+				return "", err
+			}
```

Comment:
Here we call the same function with the same parameter twice.

We could avoid this:
```suggestion
			if isRetryableFlareError(err, statusCode) {
				if attempt > 0 {
					log.Warn("Failed to send flare, retrying in 1 second")
					time.Sleep(1*time.Second)
				}
				return "", err
			} else {
				return "", err
			}
```